### PR TITLE
Isolate direct routing device functionality

### PR DESCRIPTION
**Pull request by bimmlerd**
This moves the resolving of the DirectRoutingDevice into its own component allowing us to finally remove the `DeviceManager` shim.

### CODE REVIEW

**Comment by joamaki** [comment]
ci-e2e failing due to same reason as it's failing in all the other PRs, so marking this ready for review.

File: pkg/datapath/tables/device.go [hunk 1]
```diff
@@ -174,3 +174,31 @@ func DeviceNames(devs []*Device) (names []string) {
...
+
+// NonEmpty returns true if the filter has been defined
+// (i.e. user has specified --devices).
+func (lst DeviceFilter) NonEmpty() bool {
+	return len(lst) > 0
+}
+
+// Match checks whether the given device name passes the filter
+func (lst DeviceFilter) Match(dev string) bool {
+	if len(lst) == 0 {
```
Comment (tommyp1ckles) [inline pkg/datapath/tables/device.go:190]: So no filter is any here?

File: pkg/datapath/linux/config/config.go [hunk 2]
```diff
@@ -567,23 +567,42 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
...
-			ifindex, ip, ok := h.nodeAddressing.IPv6().DirectRouting()
-			if !ok {
+			if drd == nil {
 				return fmt.Errorf("IPv6 direct routing device not found")
 			}
+
+			var ip net.IP
+			for _, addr := range drd.Addrs {
+				if addr.Addr.Is6() {
+					ip = addr.AsIP()
```
Comment (tommyp1ckles) [inline pkg/datapath/linux/config/config.go:600]: random question, why is ip6 not stored in network byte order?

**Review by tommyp1ckles — APPROVED** [review]
CLI changes look good

**Review by christarazi — APPROVED** [review]
ci-structure LGTM

**Review by joamaki — APPROVED** [review]

**Review by dylandreimerink — APPROVED** [review]
LGTM

**Pull request merged**